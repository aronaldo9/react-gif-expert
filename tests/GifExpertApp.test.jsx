/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import GifExpertApp from "../src/GifExpertApp";

// Mock de componentes dependientes
jest.mock("../src/components/AddCategory", () => () => (
  <div>AddCategory Mock</div>
));
jest.mock("../src/components/GifGrid", () => ({ category }) => (
  <div>GifGrid Mock: {category}</div>
));

describe("Pruebas en <GifExpertApp />", () => {
  test("Debe de mostrar el componente correctamente", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar una lista de categorías", () => {
    const categories = ["One Punch", "Dragon Ball"];
    render(<GifExpertApp defaultCategories={categories} />);

    // Verificar que los elementos están en el documento
    expect(screen.getByText("GifExpertApp")).toBeInTheDocument();
    expect(screen.getByText("AddCategory Mock")).toBeInTheDocument();

    // Verificar que se muestran las categorías
    expect(screen.getByText("GifGrid Mock: One Punch")).toBeInTheDocument();
    expect(screen.getByText("GifGrid Mock: Dragon Ball")).toBeInTheDocument();
  });
});
