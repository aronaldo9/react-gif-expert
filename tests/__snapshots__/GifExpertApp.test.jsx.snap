// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <div
    class="flex justify-center items-center h-32"
  >
    <div
      class="animate-spin rounded-full h-16 w-16 border-t-8 border-blue-600 border-solid"
    />
  </div>
  <div
    class="card-grid"
  />
</div>
`;
